### ISSUE TITLE
HTTP.read resolves with buffer instead of string

### ISSUE THREAD

**Issue by Tomalak** [kriskowal/q-io#139]
This

``` javascript
var qHttp = require("q-io/http");

qHttp.read("https://github.com/").then(function(data) {
    console.log(data);
}).fail(function (reason) {
    console.log("Error: " + reason);
});
```

logs `<Buffer 3c 21 44 4f 43 54 59 50 45 20 68 74 ...`. 

It _should_ log the contents of the page as a string, though. (The docs state _"returns a promise for the response body as a string"_.)

As a user of `.read()` I have no way of knowing the response charset, so there is no safe way of turning this buffer to string. 

This conversion to string must take place inside `.read()`

**Comment by Tomalak** [comment]
Pull request proposed here 

https://github.com/kriskowal/q-io/pull/140

**Comment by hthetiot** [comment]
Closed by revert I think see #145
